tenzoMath/tenzoMath.cpp

```diff
@@ -1,149 +1,521 @@
-	std::array<double, 3> newData;
-	for (int i = 0; i < 6; ++i)
-	{
-		_model.fanucForwardTask(_positions[i])(cv::Rect(0, 0, 3, 3)).copyTo(currRot);
-
-		cv::Mat gravProjection = currRot.t() * _g;
-
-		for (int j = 0; j < 3; ++j)
-		{
-			rawData[j] = _collectedData.at<double>(i, j);
-		}
-
-		/*newData[0] = rawData[0] - gravProjection.at<double>(0, 0) * _fgmax.at<double>(0, 0) - gravProjection.at<double>(1, 0)
-			* _fgmax.at<double>(0, 1) - gravProjection.at<double>(2, 0) * _fgmax.at<double>(0, 2);
-		newData[1] = rawData[1] - gravProjection.at<double>(1, 0) * _fgmax.at<double>(1, 1) - gravProjection.at<double>(0, 0)
-			* _fgmax.at<double>(1, 0) - gravProjection.at<double>(2, 0) * _fgmax.at<double>(1, 2);
-		newData[2] = rawData[2] - gravProjection.at<double>(2, 0) * _fgmax.at<double>(2, 2) - gravProjection.at<double>(0, 0)
-			* _fgmax.at<double>(2, 0) - gravProjection.at<double>(2, 0) * _fgmax.at<double>(2, 2);
-*/
-		std::cout << "New data: " << newData[0] << '\t' << newData[1] << '\t' << newData[2] << std::endl;
-	}
+    std::array<double, 6> tmp;
+    tmp[0] = data[1];
+    tmp[1] = -data[0];
+    tmp[2] = -data[2];
+    tmp[3] = -data[4];
+    tmp[4] = data[3];
+    tmp[5] = data[5];
+    return tmp;
 }
 
-void TenzoMath::gravCompensation()
+std::string TenzoMath::toString(std::array<double, 6> coord) const
 {
-	/**
-	* \brief object for connecting with fanuc
-	*/
-	FanucM20iA _fanuc;
-	_fanuc.setJointFrame();
-	Tenzo tenzoData(L"COM8");
-	std::array<double, 6> currJoints;
-	cv::Mat currRot(3, 3, CV_64F);
-	std::array<double, 6> rawData;
-	std::array<double, 3> newData;
-	while (true)
-	{
-		currJoints = _fanuc.getJointAngles();
-
-		for (int i = 0; i < 6; ++i)
-			currJoints[i] /= 1000.0;
-
-		std::cout << "Current joints: " << currJoints[0] << '\t' << currJoints[1] << '\t' << currJoints[2] << '\t' <<
-			currJoints[3] << '\t' << currJoints[4] << '\t' << currJoints[5] << std::endl;
-
-		_model.fanucForwardTask(currJoints)(cv::Rect(0, 0, 3, 3)).copyTo(currRot);
-
-		cv::Mat gravProjection = currRot.t() * _g;
-
-		rawData = tenzoData.readData();
-
-	/*	newData[0] = rawData[0] - gravProjection.at<double>(0, 0) * _fgmax.at<double>(0, 0) - gravProjection.at<double>(1, 0)
-			* _fgmax.at<double>(0, 1) - gravProjection.at<double>(2, 0) * _fgmax.at<double>(0, 2);
-		newData[1] = rawData[1] - gravProjection.at<double>(1, 0) * _fgmax.at<double>(1, 1) - gravProjection.at<double>(0, 0)
-			* _fgmax.at<double>(1, 0) - gravProjection.at<double>(2, 0) * _fgmax.at<double>(1, 2);
-		newData[2] = rawData[2] - gravProjection.at<double>(2, 0) * _fgmax.at<double>(2, 2) - gravProjection.at<double>(0, 0)
-			* _fgmax.at<double>(2, 0) - gravProjection.at<double>(2, 0) * _fgmax.at<double>(2, 2);
-*/
-		std::cout << "New data: " << newData[0] << '\t' << newData[1] << '\t' << newData[2] << std::endl;
-	}
+    std::string tmp;
+    for (int i = 0; i < 6; ++i)
+    {
+        tmp = tmp + std::to_string(coord[i]) + ' ';
+    }
+    return tmp;
```

Comment:
(**must be changed**): More efficient way of printing containters and raw C-style arrays:
```C++
// std::array<double, 6> coord; -> input parameter(!)
std::copy(std::begin(coord), std::end(coord), std::ostream_iterator<int>(std::cout, " "));
```